### PR TITLE
Restructure click shield

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Click shields

### Description
This PR:
 - removes the static fallback image from the vimeo frame. The set URL is broken.
 - adds the ability to define background colors for the click shield background
 - renames one-click-embedding to click shield

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Comment by ErikMichelson** [comment]
Do we want to use another fallback image? Has Vimeo a new URL for this for example or do we want to include such a fallback graphic in our repo?

**Comment by mrdrogdrog** [comment]
The YT frame also has no fallback image. I don't know if we need one.

**Comment by ErikMichelson** [comment]
Okay. What about adding a static background color to highlight the one-click-component from the page background when no thumbnail is available/loaded?

This background color could match the service's appearance or logo colors. For Vimeo this would be `#00ADEF`, for YouTube `#FF0000`.

**Comment by mrdrogdrog** [comment]
Sounds good to me

**Comment by mrdrogdrog** [comment]
@ErikMichelson

**Review by ErikMichelson — DISMISSED** [review]

**Comment by ErikMichelson** [comment]
Please fix the prettier issues

File: src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  loadingImageUrl?: string
+  hoverIcon?: IconName
+  hoverTextI18nKey?: string
+  targetDescription?: string
+  containerClassName?: string
+  onActivate?: () => void
+  fallbackBackgroundColor?: Property.BackgroundColor
+}
+
+export const ClickShield: React.FC<ClickShieldProps> = ({
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:31]: This needs documentation

File: src/components/markdown-renderer/replace-components/asciinema/asciinema-frame.tsx [hunk 2]
```diff
@@ -5,22 +5,24 @@
  */
 
 import React from 'react'
-import { OneClickEmbedding } from '../one-click-frame/one-click-embedding'
+import { ClickShield } from '../click-shield/click-shield'
 import type { IdProps } from '../custom-tag-with-id-component-replacer'
 
 export const AsciinemaFrame: React.FC<IdProps> = ({ id }) => {
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/asciinema/asciinema-frame.tsx:11]: Add documentation

File: src/components/markdown-renderer/replace-components/youtube/youtube-frame.tsx [hunk 3]
```diff
@@ -5,23 +5,25 @@
  */
 
 import React from 'react'
-import { OneClickEmbedding } from '../one-click-frame/one-click-embedding'
+import { ClickShield } from '../click-shield/click-shield'
 import type { IdProps } from '../custom-tag-with-id-component-replacer'
 
 export const YouTubeFrame: React.FC<IdProps> = ({ id }) => {
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/youtube/youtube-frame.tsx:11]: documentation

**Review by DerMolly — CHANGES_REQUESTED** [review]
VimeoFrame also needs documentation

File: src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  containerClassName,
+  onImageFetch,
+  fallbackPreviewImageUrl,
+  children,
+  targetDescription,
+  hoverIcon,
+  fallbackBackgroundColor,
+  ...props
+}) => {
+  const [showFrame, setShowFrame] = useState(false)
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:51]: Maybe this should be called showChildren?

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by DerMolly — APPROVED** [review]

File: src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  hoverTextI18nKey?: string
+  targetDescription?: string
+  containerClassName?: string
+  fallbackBackgroundColor?: Property.BackgroundColor
+}
+
+/**
+ * Prevents loading of the children elements until the user unlocks the content by e.g. clicking.
+ *
+ * @param containerClassName Additional CSS classes for the complete component
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:34]: Why not call this attribute `additionalClasses` like in [other](https://github.com/hedgedoc/react-client/blob/0aae1f70d20c425335421981f76cd58fc520967e/src/components/common/user-avatar/user-avatar.tsx#L16) [places](https://github.com/hedgedoc/react-client/blob/a398660c18e89489bd08f2925a61f2a54c282d8a/src/components/common/modals/common-modal.tsx#L24)?

File: src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+          className={'preview-background embed-responsive-item'}
+          style={fallbackBackgroundStyle}
+          src={previewImageUrl}
+          alt={previewHoverText}
+          title={previewHoverText}
+        />
+      )
+    } else {
+      return <span className={'preview-background embed-responsive-item'} style={fallbackBackgroundStyle} />
+    }
```
Comment (ErikMichelson) [inline src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:100]: Simplify by swapping if-else?

```suggestion
    if (!previewImageUrl) {
      return <span className={'preview-background embed-responsive-item'} style={fallbackBackgroundStyle} />
    }
    return (
      <ProxyImageFrame
        className={'preview-background embed-responsive-item'}
        style={fallbackBackgroundStyle}
        src={previewImageUrl}
        alt={previewHoverText}
        title={previewHoverText}
      />
    )
```

File: src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx (hunk 5 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:34]: Because that would tell you where the classes will be added.

Comment (ErikMichelson) [reply src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx:34]: ok

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**